lib/src/result.dart

```diff
@@ -37,7 +37,7 @@ abstract class Result<T> {
   /// The result of the transformation is a sink that only forwards [Result]
   /// values and no error events.
   static const StreamSinkTransformer<Object, Result> captureSinkTransformer =
-      const StreamSinkTransformer.fromStreamTransformer(
```

Comment:
without seeing the error msg, it's hard to say for sure. I don't think dynamic vs Object would itself be problematic because those are treated almost identically for the subtype relation. You could try `const CaptureStreamTransformer<Object>()` and see if that fixes the problem.

Perhaps also worth trying without the `const`. `const` goes down pretty different code paths in Analyzer.